SimTKcommon/tests/TestStateXml.cpp

```diff
@@ -25,6 +25,46 @@
 
 using namespace SimTK;
 
+/* This function converts the provided instance of `T` to Xml, deserializes
+it from Xml, ensures the deserialized instance is equal to the original
+instance, reserializes the deserialized instance back to Xml and ensures
+the original desrialization and the reserialization are equal.
+
+One might set `checkInMemoryEquality` to false if `value` contains NaNs. */
```

Comment:
This seems awkward, and we do want to check if NaNs go in and out correctly. Could you add explicit isNaN() checks? It would be good to make sure inf and -inf are working too (operator== is fine for them but any serialization that doesn't go properly through Simbody's formatting for floating point numbers won't work right).